modules/bmc/redfish.rb

```diff
@@ -0,0 +1,210 @@
+require 'redfish_client'
+require 'bmc/base'
+require 'bmc/redfish/dell'
+require 'bmc/redfish/hpe'
+
+module Proxy
+  module BMC
+    class Redfish < Base
+      include Proxy::Log
+      include Proxy::Util
+
+      def initialize(args)
+        @redfish_verify_ssl = Proxy::BMC::Plugin.settings.bmc_redfish_verify_ssl
+        # enforce SSL verification as a default, if the value isn't set in the config
+        # file, or is set to a "garbage" value (anything but boolean false)
+        @redfish_verify_ssl = true unless @redfish_verify_ssl == false
+        super
+        load_vendor_overrides
+        @bmc
+      end
+
+      def connect(args = { })
+        # TODO probably verify should be an option.
+        connection = RedfishClient.new("https://#{args[:host]}/", verify: @redfish_verify_ssl)
+        connection.login(args[:username], args[:password])
+        connection
+      end
+
+      def load_vendor_overrides
+        mfr = manufacturer
+        return unless mfr
+        mfr.tr!('^A-Za-z', '')
+        begin
+          mod = Kernel.const_get("RedfishVendorOverrides#{mfr}".to_sym)
+        rescue NameError
+          # no extensions for this vendor
+          return
+        end
+        logger.debug "Extending Redfish with vendor overrides for #{mfr}"
+        self.extend mod
+      end
+
+      def manufacturer
+        host.Systems.Members[0].Manufacturer
+      end
+
+      # returns boolean if the test is successful
+      def test
+        # pretty low-effort try, here. sufficient, though?
+        begin
+          host.Managers.Members.any?
+        rescue NoMethodError
+          false
+        end
+      end
+
+      def identifystatus
+        host.Systems.Members[0].IndicatorLED.downcase
```

Comment:
`host.Systems.Members[0]` is repeated here a lot, maybe it would be worth extracting to a separate method?

Reply:
Also check for the boundaries in that method and error out nicely.